omegaconf/basecontainer.py

```diff
@@ -193,10 +193,16 @@ def _to_content(
         resolve: bool,
         enum_to_str: bool = False,
         exclude_structured_configs: bool = False,
+        instantiate: bool = False,
     ) -> Union[None, Any, str, Dict[DictKeyType, Any], List[Any]]:
         from .dictconfig import DictConfig
         from .listconfig import ListConfig
 
+        if exclude_structured_configs and instantiate:
+            raise ValueError(
+                "Cannot both exclude and and instantiate structured configs"
+            )
+
```

Comment:
exclude_structured_configs is new in 2.1 and we can still change it.
Looks like an enum can replace exclude_structured_configs and instantiate:
```python
class SCMode(Enum):
  DICT = 1            # convert to plain dict
  DICT_CONFIG = 2     # Keep as OmegaConf DictConfig
  INSTANTIATE = 3     # instantiate underlying object.
```

Reply:
Looks good.